File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.java
class Solution {
    public int longestSubsequence(int[] nums) {

        int n = nums.length;

        int xor = 0;
        boolean hasNonZero = false;

        for (int x : nums) {
            xor ^= x;

            if (x != 0)
                hasNonZero = true;
        }

        // Entire array works
        if (xor != 0)
            return n;

        // Every element is zero
        if (!hasNonZero)
            return 0;

        // Remove one non-zero element
        return n - 1;
    }
}

File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.c
int longestSubsequence(int* nums, int numsSize) {

    int xor = 0;
    int hasNonZero = 0;

    for (int i = 0; i < numsSize; i++) {

        xor ^= nums[i];

        if (nums[i] != 0) {
            hasNonZero = 1;
        }
    }

    // Entire array works
    if (xor != 0) {
        return numsSize;
    }

    // All elements are zero
    if (!hasNonZero) {
        return 0;
    }

    // Remove one non-zero element
    return numsSize - 1;
}